radixdlt-core/radixdlt/src/main/java/com/radixdlt/consensus/epoch/EpochManager.java

```diff
@@ -312,63 +310,69 @@ private void processEpochChange(EpochChange epochChange) {
       }
     }
 
+    final var queuedEventsForNewEpoch = queuedEventsForNextEpoch;
+    queuedEventsForNextEpoch = new ArrayList<>(256);
+
     this.currentEpoch = epochChange;
     this.updateEpochState();
     this.bftEventProcessor.start();
 
-    // Execute any queued up consensus events
-    final List<ConsensusEvent> queuedEventsForEpoch =
-        queuedEvents.getOrDefault(epochChange.getEpoch(), Collections.emptyList());
-    var highView =
-        queuedEventsForEpoch.stream()
-            .map(ConsensusEvent::getView)
-            .max(Comparator.naturalOrder())
-            .orElse(View.genesis());
-
-    queuedEventsForEpoch.stream()
-        .filter(e -> e.getView().equals(highView))
-        .forEach(this::processConsensusEventInternal);
-
-    queuedEvents.remove(epochChange.getEpoch());
+    this.processCachedConsensusEventsAtStartOfEpoch(epochChange, queuedEventsForNewEpoch);
   }
 
-  private void processConsensusEventInternal(ConsensusEvent consensusEvent) {
-    this.counters.increment(CounterType.BFT_EVENTS_RECEIVED);
+  public void processConsensusEvent(ConsensusEvent consensusEvent) {
+    final var currentEpoch = this.currentEpoch();
 
-    switch (consensusEvent) {
-      case Proposal proposal -> bftEventProcessor.processProposal(proposal);
-      case Vote vote -> bftEventProcessor.processVote(vote);
+    if (consensusEvent.getEpoch() == currentEpoch) {
```

Comment:
I would use a `if`, `else if` and `else` in this function instead of 3 ifs.

Reply:
That's definitely a semantically correct choice. Although, to explain - this was a purposeful choice to follow the "early return" pattern - https://dev.to/jpswade/return-early-12o5

In general, I aim to reduce nesting and the amount of stuff you need to remember, and I typically avoid else blocks, favouring taking out methods and early returns; which I (personally) find to be a pattern which is more generally applicable and easier to read.